src/main/java/com/mitchej123/hodgepodge/mixins/early/minecraft/MixinFontRenderer.java

```diff
@@ -40,10 +41,9 @@ public String wrapFormattedStringToWidth(String str, int wrapWidth) {
                 break;
             }
             output.append('\n');
-            output.append(getFormatFromString(line));
             final char nextChar = str.charAt(lineWidth);
             final boolean nextIsBlank = nextChar == ' ' || nextChar == '\n';
-            str = StringUtils.substring(str, lineWidth + (nextIsBlank ? 1 : 0));
+            str = getFormatFromString(line) + StringUtils.substring(str, lineWidth + (nextIsBlank ? 1 : 0));
```

Comment:
Can you remove the `+` concatenation/allocation by storing getFormatFromString in a variable outside the loop, and use `output.append()` to insert it in the correct place instead? This is somewhat performance-sensitive code, one of the goals here is to reduce memory allocations as much as possible (substring does not create new string data arrays, while + does).